src/main/java/org/broadinstitute/hellbender/tools/exome/AllelicCountCollection.java

```diff
@@ -87,24 +79,85 @@ public void add(final SimpleInterval interval, final int refReadCount, final int
+                new TableColumnCollection(AllelicCountTableColumns.BASIC_COLUMN_NAME_ARRAY),
+                (count, dataLine) -> {
+                    final SimpleInterval interval = count.getInterval();
+                    final int refReadCount = count.getRefReadCount();
+                    final int altReadCount = count.getAltReadCount();
+                    dataLine.append(interval.getContig())
+                            .append(interval.getEnd())
+                            .append(refReadCount)
+                            .append(altReadCount);
+                })) {
+            writer.writeAllRecords(counts);
+        } catch (final IOException e) {
+            throw new UserException.CouldNotCreateOutputFile(outputFile, e);
+        }
+    }
+
+    /**
+     * Writes out intermediate pulldown data (sequence, position, reference count, alternate count, reference nucleotide,
+     * alternate nucleotide) to specified file.
+     * @param outputFile    file to write to (if it exists, it will be overwritten)
+     */
+    public void writeIntermediateCollection(final File outputFile) {
+        try (final TableWriter<AllelicCount> writer = TableUtils.writer(outputFile,
+                new TableColumnCollection(AllelicCountTableColumns.INTERMEDIATE_COLUMN_NAME_ARRAY),
+                (count, dataLine) -> {
+                    final SimpleInterval interval = count.getInterval();
+                    final int refReadCount = count.getRefReadCount();
+                    final int altReadCount = count.getAltReadCount();
+                    final Nucleotide refNucleotide = count.getRefNucleotide();
+                    final Nucleotide altNucleotide = count.getAltNucleotide();
+                    final int readDepth = count.getReadDepth();
+                    dataLine.append(interval.getContig())
+                            .append(interval.getEnd())
+                            .append(refReadCount)
+                            .append(altReadCount)
+                            .append(refNucleotide.name())
+                            .append(altNucleotide.name())
+                            .append(readDepth);
+                })) {
+            writer.writeAllRecords(counts);
+        } catch (final IOException e) {
+            throw new UserException.CouldNotCreateOutputFile(outputFile, e);
+        }
     }
 
     /**
-     * Writes out (sequence, position, reference count, alternate count) to specified file.
+     * Writes out full pulldown data (sequence, position, reference count, alternate count, reference nucleotide,
+     * alternate nucleotide, het log odds) to specified file.
      * @param outputFile    file to write to (if it exists, it will be overwritten)
      */
-    public void write(final File outputFile) {
+    public void writeFullCollection(final File outputFile) {
         try (final TableWriter<AllelicCount> writer = TableUtils.writer(outputFile,
```

Comment:
Don't need to create a separate for each type of output (FULL, INTERMEDIATE,) you can make the class choose between either mode based on some constructor parameter.